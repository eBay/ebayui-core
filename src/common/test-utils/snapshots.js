import { render, prettyDOM } from '@marko/testing-library';
import snap from 'mocha-snap';

function snapshotNode(dirname) {
    return (container) => snap(container, '.html', dirname);
}

function snapshotHTML(dirname) {
    const localSnap = snapshotNode(dirname);
    return async (template, input) => {
        const { container } = await render(template, input);
        await localSnap(prettyDOM(container));
    };
}

export { snapshotHTML, snapshotNode };
