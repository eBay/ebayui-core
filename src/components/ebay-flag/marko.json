{
    "taglib-id": "ebayui-core-flags",
    "tags-dir": "./icons"
}
