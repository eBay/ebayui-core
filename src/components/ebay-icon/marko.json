{
  "taglib-id": "ebayui-core-icons",
  "tags-dir": "./icons"
}