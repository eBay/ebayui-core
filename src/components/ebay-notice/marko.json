{
    "<ebay-notice>": {
        "renderer": "./index.js"
    }
}
