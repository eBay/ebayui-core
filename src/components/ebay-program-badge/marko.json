{
  "taglib-id": "ebayui-core-program-badges",
  "tags-dir": "./icons"
}
