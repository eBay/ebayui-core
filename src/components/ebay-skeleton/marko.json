{
    "taglib-id": "ebayui-core-skeleton",
    "tags-dir": "./components"
}
