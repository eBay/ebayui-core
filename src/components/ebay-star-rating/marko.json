{
    "taglib-id": "ebayui-core-star-rating",
    "tags-dir": "./icons"
}
