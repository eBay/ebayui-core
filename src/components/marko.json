{
  "<ebay-tooltip-base>": {
    "renderer": "./components/ebay-tooltip-base/index.js"
  },
  "<ebay-tooltip-overlay>": {
    "renderer": "./components/ebay-tooltip-overlay/index.js"
  },
  "<*>": {
    "transformer": "../common/transformers/ebayui-attribute/index.js"
  }
}