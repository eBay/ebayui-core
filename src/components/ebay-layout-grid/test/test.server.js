import { it } from "vitest";
import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../layout-button.stories"; // import all stories from the stories file

const { Default, WithCustomColumns } = composeStories(stories);
const htmlSnap = snapshotHTML(__dirname);

describe("ebay-layout-grid", () => {
    it("renders default", async () => {
        await htmlSnap(Default);
    });

    it("renders with custom columns", async () => {
        await htmlSnap(WithCustomColumns);
    });
});
