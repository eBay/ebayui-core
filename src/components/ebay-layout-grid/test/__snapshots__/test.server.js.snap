// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ebay-layout-grid > renders default 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33mclass[39m=[32m"layout-grid"[39m
    [33mdata-columns-lg[39m=[32m"6"[39m
    [33mdata-columns-md[39m=[32m"4"[39m
    [33mdata-columns-min[39m=[32m"1"[39m
    [33mdata-columns-sm[39m=[32m"3"[39m
    [33mdata-columns-xl[39m=[32m"8"[39m
    [33mdata-columns-xl2[39m=[32m"10"[39m
    [33mdata-columns-xl3[39m=[32m"12"[39m
    [33mdata-columns-xl4[39m=[32m"14"[39m
    [33mdata-columns-xs[39m=[32m"2"[39m
  [36m>[39m
    [36m<ul>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[0]-1 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[0]-1 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[0]-1 false\\",\\"onblur\\":\\"handleBlur s0-0-3[0]-1 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 1[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[1]-2 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[1]-2 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[1]-2 false\\",\\"onblur\\":\\"handleBlur s0-0-3[1]-2 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 2[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[2]-3 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[2]-3 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[2]-3 false\\",\\"onblur\\":\\"handleBlur s0-0-3[2]-3 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 3[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[3]-4 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[3]-4 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[3]-4 false\\",\\"onblur\\":\\"handleBlur s0-0-3[3]-4 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 4[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[4]-5 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[4]-5 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[4]-5 false\\",\\"onblur\\":\\"handleBlur s0-0-3[4]-5 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 5[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[5]-6 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[5]-6 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[5]-6 false\\",\\"onblur\\":\\"handleBlur s0-0-3[5]-6 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 6[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[6]-7 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[6]-7 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[6]-7 false\\",\\"onblur\\":\\"handleBlur s0-0-3[6]-7 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 7[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[7]-8 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[7]-8 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[7]-8 false\\",\\"onblur\\":\\"handleBlur s0-0-3[7]-8 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 8[0m
        [36m</button>[39m
      [36m</li>[39m
    [36m</ul>[39m
  [36m</div>[39m
[36m</DocumentFragment>[39m"
`;

exports[`ebay-layout-grid > renders with custom columns 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33mclass[39m=[32m"layout-grid"[39m
    [33mdata-columns-lg[39m=[32m"8"[39m
    [33mdata-columns-md[39m=[32m"6"[39m
    [33mdata-columns-min[39m=[32m"2"[39m
    [33mdata-columns-sm[39m=[32m"4"[39m
    [33mdata-columns-xs[39m=[32m"3"[39m
  [36m>[39m
    [36m<ul>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[0]-1 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[0]-1 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[0]-1 false\\",\\"onblur\\":\\"handleBlur s0-0-3[0]-1 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 1[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[1]-2 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[1]-2 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[1]-2 false\\",\\"onblur\\":\\"handleBlur s0-0-3[1]-2 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 2[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[2]-3 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[2]-3 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[2]-3 false\\",\\"onblur\\":\\"handleBlur s0-0-3[2]-3 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 3[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[3]-4 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[3]-4 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[3]-4 false\\",\\"onblur\\":\\"handleBlur s0-0-3[3]-4 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 4[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[4]-5 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[4]-5 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[4]-5 false\\",\\"onblur\\":\\"handleBlur s0-0-3[4]-5 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 5[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[5]-6 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[5]-6 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[5]-6 false\\",\\"onblur\\":\\"handleBlur s0-0-3[5]-6 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 6[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[6]-7 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[6]-7 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[6]-7 false\\",\\"onblur\\":\\"handleBlur s0-0-3[6]-7 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 7[0m
        [36m</button>[39m
      [36m</li>[39m
      [36m<li>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--secondary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mdata-marko[39m=[32m"{\\"onclick\\":\\"handleClick s0-0-3[7]-8 false\\",\\"onkeydown\\":\\"handleKeydown s0-0-3[7]-8 false\\",\\"onfocus\\":\\"handleFocus s0-0-3[7]-8 false\\",\\"onblur\\":\\"handleBlur s0-0-3[7]-8 false\\"}"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mButton 8[0m
        [36m</button>[39m
      [36m</li>[39m
    [36m</ul>[39m
  [36m</div>[39m
[36m</DocumentFragment>[39m"
`;
